### ISSUE TITLE
Class-scoped generic bounded by NamedTuple not inferred correctly

### ISSUE THREAD

**Issue by delta1f** [astral-sh/ty#3907]
### Summary

When I use a protocol that is generic in a NamedTuple-bounded type parameter, and I pass that type parameter through another generic function, I encounter an invalid-argument-type error. The type checker doesn't realize that the class-scoped generic parameter should be inferred as equal to the function-scoped one. I am unsure if this is a variance issue, or a bug due to the generic parameter being class-scoped, or a bug with the NamedTuple class, or if I simply missed something :).
Minimal example in playground: https://play.ty.dev/cc99a962-9ab6-43dd-b402-2461c516ca96

### Version

0.0.55

**Comment by sharkdp** [comment]
Thank you for reporting this.

This looks like a bug in ty related to typevar bounds that are intersections (we treat `NamedTuple` like `tuple[object, ...] & NamedTupleLike`). This can also be reproduced without NamedTuple:

```py
from typing import Protocol

class P: ...
class Q: ...

class Creator[Output: P & Q](Protocol):
    special_output: Output
    def create(self) -> Output: ...


def run_creator_fn[Output: P & Q](creator: Creator[Output]) -> Output:
    return creator.create()
```

Potentially related issue with typevar bounds that are *unions*: https://github.com/astral-sh/ty/issues/2585

**Comment by delta1f** [comment]
@sharkdp It seems this was fixed by [#26701](https://github.com/astral-sh/ruff/pull/26701).
